Explore: Logging line parsing and field stats

davkal opened the pull request:
Lazily parse lines and display stats for field when clicked on.

- lazy parsing to not delay result display
- active line (mouseover) determines parser, in case various streams are mixed
- log line parsers for JSON (basic), and logfmt
- delay-and-cancel parsing (500ms) in case user moves to other line
- reuse label stats for field stats

For later: improve JSON value type parsing (currently only strings are supported)

davkal committed:
Explore: Logging line parsing and field stats

Lazily parse lines and display stats for field when clicked on.

- log line parsers for JSON (basic), and logfmt
- delayed parsing in case user moves to other line
- reuse label stats for field stats

torkelo reviewed (changes requested):
code looks ok, the styling for expanded state was a bit broken due the grid overflow, but the close button becomes a bit lost. Feel the expand styles needs to be a bit clearer, some kind of box with margin or something to offset it. The heading row of the stats "box" just under log line and looks like it's part of log  message. 

We can start making the initial parsing on click instead of hover. as only leaving mouse causes things to jump around sometimes.

davkal committed:
Rebase fixes

davkal commented:
- rebased, fixed styles
- styled box around row field stats
- grid overflow is no longer happening
- not getting much jank anymore, I'm inclined to keep mouseover parse for now, but I'm happy to change it to onClick

@torkelo PTAL

torkelo committed:
Aligned styling of stats popover/box with rest of grafana & minor css refactoring

torkelo reviewed (changes requested):
* Aligned the styling with our other similar popover
* Added max width so when used in the message body it does not cover whole window width. Think popover makes more sense here as it is for labels. but not important. 

![image](https://user-images.githubusercontent.com/10999/49642533-9d965d80-fa13-11e8-8599-c0b16931087f.png)


![image](https://user-images.githubusercontent.com/10999/49642494-86577000-fa13-11e8-8940-ee06c2e5e633.png)

TODO
Logs.tsx needs some refactoring breaking up.

* Move Row to it's own file (name it LogRow or similar) (can then rename props to Props & State). 
* Move Stats to it's own file (name it LogRowStats or similar if you can come up with a more descriptive name) (in LogLabels.tsx now)

davkal commented:
LGTM
